src/app/services/views/ServiceListView.vue

```diff
@@ -124,13 +133,33 @@
                           },
                           label: t('common.collection.actions.view'),
                         }"
+                        data-testid="dropdown-view-details-item"
                       />
                     </template>
                   </KDropdownMenu>
                 </template>
               </AppCollection>
             </template>
           </KCard>
+
+          <template
+            v-for="(selectedService, index) in [data?.items.find((item) => item.name === route.params.service)]"
+            :key="index"
+          >
+            <RouterView
+              v-if="selectedService"
+              v-slot="child"
+            >
+              <component
+                :is="child.Component"
+                :data="selectedService"
+                @tray-close="route.update({
+                  page: route.params.page,
+                  size: route.params.size,
+                })"
+              />
+            </RouterView>
+          </template>
```

Comment:
All sub views of a listing view will be trays right? Why not just put the Tray component here? (I'd say it almost belongs here in the listing view in the way that they are both super related, i.e. you need the list to see the tray)

I think this would also make the code clearer (at the moment its hard to see that this is a more or less a tray component). You also wouldn't need to do the extra plumbing we have in the ServiceTrayView to pass through `@close`.

Another benefit is that in the cases that some trays might be the same as the detail view, we could potentially avoid multiple/doubling route components for essentially the same content. i.e. we could just reuse the detail view inside of the tray. I think from our sync the other day we said that it is probably quite common that the content in the tray and the detail view is exactly the same.

Reply:
> All sub views of a listing view will be trays right? Why not just put the Tray component here? (I'd say it almost belongs here in the listing view in the way that they are both super related, i.e. you need the list to see the tray)

That’s because we hook up ServiceTrayView via its tray route which means ServiceListView must have a `RouterView` rather than a ServiceTray component directly.

I’m not sure there’s an alternative here.

> Another benefit is that in the cases that some trays might be the same as the detail view, we could potentially avoid multiple/doubling route components for essentially the same content. i.e. we could just reuse the detail view inside of the tray. I think from our sync the other day we said that it is probably quite common that the content in the tray and the detail view is exactly the same.

I don’t know how common this actually will be. Re-using the detail **view** components for the summary trays is problematic because their similarity is incidental and wouldn’t always work. I rather extract non-view components like ExternalServiceDetails and ServiceInsightDetails for easy re-use. Additionally, the layout of the detail view compared to the summary view might be sufficiently different to prevent re-use in the respective other.

Reply:
> That’s because we hook up ServiceTrayView via its tray route which means ServiceListView must have a RouterView rather than a ServiceTray component directly.

Yeah I getcha, but can we put the tray inside the RouteView but wrapping the `<component />`? That way all sub views of the listing are trays, and we don't have to repeat the tray for every single sub route in its *View.vue file (and the tray not being in the *View.vue file means its possible to reuse the DetailVue.vue file without much trouble)

> I don’t know how common this actually will be. Re-using the detail view components for the summary trays is problematic because their similarity is incidental and wouldn’t always work. I rather extract non-view components like ExternalServiceDetails and ServiceInsightDetails for easy re-use. Additionally, the layout of the detail view compared to the summary view might be sufficiently different to prevent re-use in the respective other.

Yeah maybe I suppose we will see, but pretty sure we said that them both being the same (or even very similar) would be reasonably common.

Leaving View reuse aside, overall on this point (whether we reuse the DetailView.vue's or not), I think its worth putting the tray inside the `<RouterView>` but wrapping the `<component>`. Does what I'm saying make more sense now? From reading your response, its sounds like I'd not explained properly and you'd understood that I was suggesting that the tray view would be _instead of_ the RouterView, which is not what I meant. lmk

Reply:
> Yeah I getcha, but can we put the tray inside the RouteView but wrapping the `<component />`? That way all sub views of the listing are trays, and we don't have to repeat the tray for every single sub route in its *View.vue file

Ah, now I understand. You want me to put the TrayView content except its default slot content into the list view. I’ve done that now.

> (and the tray not being in the *View.vue file means its possible to reuse the DetailVue.vue file without much trouble)

That’s something we just shouldn’t do. The detail views are **generally** not the same as the summary content. I don’t even want to try making them the same by re-using the views even that might be possible in some cases. In fact, it’s already not a 1-to-1 equivalence for services because for external services we show the YAML in the summary but not the detail. Much easier to re-use non-view components here.

Reply:
I’m resolving this now. The TrayView use was moved to the list view.